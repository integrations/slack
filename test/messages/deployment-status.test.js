const { DeploymentStatus } = require('../../lib/messages/deployment-status');
const deploymentStatusPendingFixture = require('../fixtures/webhooks/deployment/status_pending.json');
const deploymentStatusSuccessFixture = require('../fixtures/webhooks/deployment/status_success.json');

describe('Deployment status rendering', () => {
  test('works for pending status', () => {
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus: deploymentStatusPendingFixture.deployment_status,
      deployment: deploymentStatusPendingFixture.deployment,
      repository: deploymentStatusPendingFixture.repository,
    });
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });

  test('works for success status', () => {
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus: deploymentStatusSuccessFixture.deployment_status,
      deployment: deploymentStatusSuccessFixture.deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
  test('works for failure status', () => {
    const deploymentStatus = {
      ...deploymentStatusSuccessFixture.deployment_status,
      state: 'failure',
    };
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus,
      deployment: deploymentStatusSuccessFixture.deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
  test('works for in_progress status', () => {
    const deploymentStatus = {
      ...deploymentStatusSuccessFixture.deployment_status,
      state: 'in_progress',
    };
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus,
      deployment: deploymentStatusSuccessFixture.deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
  test('works for queued status', () => {
    const deploymentStatus = {
      ...deploymentStatusSuccessFixture.deployment_status,
      state: 'queued',
    };
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus,
      deployment: deploymentStatusSuccessFixture.deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
  test('works for error status', () => {
    const deploymentStatus = {
      ...deploymentStatusSuccessFixture.deployment_status,
      state: 'error',
    };
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus,
      deployment: deploymentStatusSuccessFixture.deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
  test('works for statuses without a target URL', () => {
    const deploymentStatus = {
      ...deploymentStatusSuccessFixture.deployment_status,
      target_url: undefined,
    };
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus,
      deployment: deploymentStatusSuccessFixture.deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
  test('uses ref and the short sha, when ref differs from the sha', () => {
    const deployment = {
      ...deploymentStatusSuccessFixture.deployment,
      ref: 'master',
    };
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus: deploymentStatusSuccessFixture.deployment_status,
      deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });

    const message = deploymentStatusMessage.toJSON().attachments[0].text;
    expect(message).toMatch(/successfully deployed.*master.*\(3dce5c8\)/i);
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
  test('uses short sha if ref is unequal but null', () => {
    const deployment = {
      ...deploymentStatusSuccessFixture.deployment,
      ref: null,
    };
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus: deploymentStatusSuccessFixture.deployment_status,
      deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });

    const message = deploymentStatusMessage.toJSON().attachments[0].text;
    expect(message).toMatch(/successfully deployed.*3dce5c8.*/i);
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
});
