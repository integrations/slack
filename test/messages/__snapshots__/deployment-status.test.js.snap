// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deployment status rendering uses ref and the short sha, when ref differs from the sha 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#28a745",
      "fallback": "[github-slack/app] Successfully deployed master (3dce5c8) to github-slack-app",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Successfully deployed <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`master (3dce5c8)\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
    },
  ],
}
`;

exports[`Deployment status rendering uses short sha if ref is unequal but null 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#28a745",
      "fallback": "[github-slack/app] Successfully deployed 3dce5c8 to github-slack-app",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Successfully deployed <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
    },
  ],
}
`;

exports[`Deployment status rendering works for error status 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#cb2431",
      "fallback": "[github-slack/app] Error when deploying 3dce5c8 to github-slack-app",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Error when deploying <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
    },
  ],
}
`;

exports[`Deployment status rendering works for failure status 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#cb2431",
      "fallback": "[github-slack/app] Deploying 3dce5c8 to github-slack-app failed",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Deploying <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app> failed",
    },
  ],
}
`;

exports[`Deployment status rendering works for in_progress status 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#dbab09",
      "fallback": "[github-slack/app] Deployment in progress for 3dce5c8 to github-slack-app",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Deployment in progress <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
    },
  ],
}
`;

exports[`Deployment status rendering works for pending status 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#dbab09",
      "fallback": "[github-slack/app] Deploying 3dce5c8 to github-slack-app",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Deploying <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app|github-slack-app>",
    },
  ],
}
`;

exports[`Deployment status rendering works for queued status 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#dbab09",
      "fallback": "[github-slack/app] Deployment queued for 3dce5c8 to github-slack-app",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Deployment queued <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
    },
  ],
}
`;

exports[`Deployment status rendering works for statuses without a target URL 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#28a745",
      "fallback": "[github-slack/app] Successfully deployed 3dce5c8 to github-slack-app",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Successfully deployed <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to github-slack-app",
    },
  ],
}
`;

exports[`Deployment status rendering works for success status 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#28a745",
      "fallback": "[github-slack/app] Successfully deployed 3dce5c8 to github-slack-app",
      "footer": "<https://github.com/github-slack/app|github-slack/app>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Successfully deployed <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
    },
  ],
}
`;
